tests/conftest.py

```diff
@@ -580,6 +583,37 @@ def profiler_flags(runtime: str, profiler_type: str) -> List[str]:
     return flags
 
 
+@fixture
+def make_profiler_instance(
+    profiler_flags: List[str], monkeypatch: MonkeyPatch, output_directory: Path, profiler_state: ProfilerState
+) -> Callable[[], _GeneratorContextManager]:
+    """
+    Deliver an instance of profiler selected by runtime and profiler_type.
+    """
+
+    # Reuse gProfiler flow to initialize specific profiler instance - one selected by runtime fixture.
+    @contextmanager
+    def _make_profiler_instance() -> Iterator[ProfilerBase]:
```

Comment:
What's the reason to create profiler instances in such implicit way and not directly as we did before?

Personally I prefer the explicit way. This new flow is harder to understand and is more error prone :/